### ISSUE TITLE
JavadocParagraph false violation when <p> is inside of <div></div>

### ISSUE THREAD

**Issue by SteLeo1602** [checkstyle/checkstyle#16630]
I have read check documentation: https://checkstyle.org/checks/javadoc/javadocparagraph.html#JavadocParagraph
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java
/var/tmp $ cat config.xml
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
		  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
		  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocParagraph"/>
  </module>
</module>
```

/var/tmp $ cat Test.java
```
/**
 * <div>
 * This is the description.
 *
 * <p>
 * This is the paragraph.
 * </p>
 *
 * </div>
 *
 */
class Test {
}
```

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml Test.java
```
Starting audit...
[ERROR] D:\Projects\work_checkstyle\Test.java:5:4: <p> tag should be preceded with an empty line. [JavadocParagraph]
Audit done.
Checkstyle ends with 1 errors.

```


---

Expect the check to pass <p> inside <div> </div> tags since <p> is already preceded by the empty line.

---

**Comment by romani** [comment]
@Zopsss , do you remember is there was any nuance on such nesting?

**Comment by romani** [comment]
>ATTENTION:
This Check ignores HTML comments.
**The Check ignores all the nested paragraph tags**, it will not give any kind of violation if the paragraph tag is nested.

**Comment by Zopsss** [comment]
> do you remember is there was any nuance on such nesting?

tbh I don't remember as it has been so long since I worked on this check. Though I remember there was some bug in this check that we missed to fix during GSoC but I don't know if this was the that same bug or not.

**Comment by Zopsss** [comment]
> Expect the check to pass `<p>` inside `<div> </div>` tags since `<p>` is already preceded by the empty line.

Yup the `<p>` is preceded by an empty line plus it is nested paragraph so no violations are expected.

> There should be violation even there no empty line, as nested paragraphs are ignored.

I'm assuming you're trying to say `There should no violation expected even if the empty line present was not present`


---

This seems like a false-positive that we missed to fix during GSoC. Nice catch @SteLeo1602  :D

**Comment by Aziz-755** [comment]
The statement:
> The Check ignores all the nested paragraph tags, it will not give any kind of violation if the paragraph tag is nested.

seems a bit ambiguous in the documentation.
Looking at the code 

![Image](https://github.com/user-attachments/assets/ead47db1-88d9-4715-bbfa-fb23926d4254)

It is clear that "nested" refers to a `<p>` tag inside another `<p>`

However, from the discussion here, it seems like "nested" could also refer to a `<p>` tag inside any supported block-level HTML tags 
```
<address>,<blockquote> ,<div>,<dl> ,<h1>,<h2>,<h3>,<h4>,<h5>,<h6>,<hr> ,<ol>,<p>,<pre> ,<table>,<ul>
``` 

Could you clarify what the intended behavior is? Should "nested paragraphs" refer only to `<p>` inside `<p>`, or does it also include `<p>` inside other block tags?

**Comment by romani** [comment]
`<p>` inside other any tags

**Comment by Aziz-755** [comment]
I'm on it
